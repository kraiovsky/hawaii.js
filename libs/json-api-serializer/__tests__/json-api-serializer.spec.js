const { Serializer } = require('../')
const {
  queries: { mockFirstRecord },
  fixtures: { mockUsersDB },
} = require('@hypefight/test-helpers')
const mockUserObject = require('../fixtures/userSerializationObject')

const mockUserProfile = mockFirstRecord(mockUsersDB)

describe('Serializer()', () => {
  test('👍 should return serialized user with meta', async () => {
    const res = await Serializer('user', mockUserObject, mockUserProfile, { meta: 'object' })
    expect(res).toMatchSnapshot()
  })

  test('👎 should return empty user and empty meta', async () => {
    const emptyUser = {}
    const res = await Serializer('user', mockUserObject, emptyUser)
    expect(res).toMatchSnapshot()
  })
})
