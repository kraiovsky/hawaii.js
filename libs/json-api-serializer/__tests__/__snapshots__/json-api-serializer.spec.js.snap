// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serializer() 👍 should return serialized user with meta 1`] = `"{\\"jsonapi\\":{\\"version\\":\\"1.0\\"},\\"meta\\":{\\"meta\\":\\"object\\"},\\"links\\":{\\"index\\":\\"/users\\"},\\"data\\":{\\"type\\":\\"user\\",\\"id\\":\\"0ae76c84-d939-456d-bb35-f60b7659ed6b\\",\\"attributes\\":{\\"email\\":\\"email2@domain.com\\",\\"role\\":\\"user\\"},\\"links\\":{\\"self\\":\\"/users/0ae76c84-d939-456d-bb35-f60b7659ed6b\\"}}}"`;

exports[`Serializer() 👎 should return empty user and empty meta 1`] = `"{\\"jsonapi\\":{\\"version\\":\\"1.0\\"},\\"links\\":{\\"index\\":\\"/users\\"},\\"data\\":null}"`;
